ext/RMagick/rmimage.c

```diff
@@ -4388,7 +4859,11 @@ Image_density(VALUE self)
 
     image = rm_check_destroyed(self);
 
+#if defined(IMAGEMAGICK_7)
+    sprintf(density, "%gx%g", image->resolution.x, image->resolution.y);
```

Comment:
We could do that with a define but at some point the IM6 code would be remove and then we still need to change it. Not a big fan of using an alias here but I will let you decide?

Reply:
We will not make any changes to the API of ImageMagick 6. The API for IM6 is frozen and we will only do bug fixes.

Reply:
`image->resolution.x, , image->resolution.x, image->resolution.y` would become something like `RESOLUTION_X(image), RESOLUTION_Y(image)`. This does not really improve the readability. But you decide?

Reply:
Hmm, yeah, that's not ideal. I was hoping there might be a way to monkey patch `image` to have a `resolution` property against IM6 so that we could call `image->resolution.x` for both cases, but it doesn't sound like it. I guess let's leave it like this for now. I may revisit as I get a better understanding of C.